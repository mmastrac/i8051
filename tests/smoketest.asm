; 8051 test case - output byte 1 to memory 0x8000

.module smoketest
.area CSEG (CODE)

_start:
    ; Set up DPTR to point to 0x8000
    mov DPTR, #0x8000

    ; Load accumulator with value 1
    mov A, #0x01

    ; Store accumulator to external memory at DPTR
    movx @DPTR, A

    ; Halt execution
halt:
    sjmp halt

.area DSEG (DATA)
.area XSEG (XDATA)
.area HOME (CODE)
.area PSEG (CODE)

