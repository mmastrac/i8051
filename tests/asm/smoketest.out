CPU halted at 0x0006 after 4 instructions
  A=01  B=00  DPTR=8000  C=false OV=false AC=false F0=false
  R0=00 R1=00 R2=00 R3=00 R4=00 R5=00 R6=00 R7=00 
MEM: 01 00 00 00
